datalad/support/gitrepo.py

```diff
@@ -671,21 +674,174 @@ def git_remote_update(self, name='', verbose=False):
         self._git_custom_command('', 'git remote %s update %s' % (name, v),
                                  expect_stderr=True)
 
-    def git_fetch(self, name, options=''):
+    # TODO: centralize all the c&p code in fetch, pull, push
+    # TODO: document **kwargs passed to gitpython
+    def fetch(self, remote=None, refspec=None, progress=None, all=False, **kwargs):
```

Comment:
that is all cool, but let me then just repeat the question: `what is progress and all here?` (note that all overloads builtin keyword, so use of e.g. `all_` is generally advised)  No need to answer in here as a comment -- better just paste it into the docstring